sdk/storage/azfile/file/client.go

```diff
@@ -569,6 +569,7 @@ func (f *Client) DownloadFile(ctx context.Context, file *os.File, o *DownloadFil
 			return 0, err
 		}
 		size = *props.ContentLength - do.Range.Offset
+		do.Range.Count = size
```

Comment:
Do we also need to make this change in azdatalake?

Reply:
azdatalake uses the DownloadFile implemented in AzBlob. So, change is not required there